mmv1/products/cloudworkstations/api.yaml

```diff
@@ -0,0 +1,177 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: CloudWorkstations
+display_name: Cloud Workstations
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://workstations.googleapis.com/v1beta/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Cloud Workstations API
+    url: https://console.cloud.google.com/apis/library/workstations.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: 'WorkstationClusters'
+    base_url: projects/{{project}}/locations/{{location}}/workstationClusters
+    create_url: projects/{{project}}/locations/{{location}}/workstationClusters?workstationClusterId={{name}}
+    update_verb: :PATCH
+    description: |
+      A grouping of WorkstationConfigs and their associated Workstations in a region.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/workstations/docs/'
+      api: 'https://cloud.google.com/workstations/docs/reference/rest/v1beta/projects.locations.workstationClusters'
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+        timeouts: !ruby/object:Api::Timeouts
+          insert_minutes: 60
+          update_minutes: 60
+          delete_minutes: 60
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: true
+        allowed:
+          - true
+          - false
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'location'
+        description: 'The [Cloud Workstation location](https://cloud.google.com/workstations/docs/locations) for the cluster.'
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
```

Comment:
`name` from the api appears to be long-form. We should either split into an output-only `name` property and an input-only `workstationClusterId` parameter or use a custom flattener and diff suppress func.